### DISCUSSION TITLE
Suspend Workflow is not working as expected

### DISCUSSION THREAD

**Discussion by Anjan4711** [danielgerlag/workflow-core#968] [Q&A]
a workflow has step: a,b,c & d;
at first, get workflowId by StartWorkflow() such as
...
_workflowId = await _host.StartWorkflow(_workflowName, data);
//_workflowId value is "03549b82-4cc7-43dd-b927-de2e1ae75711"
...

and the workflow keeps running , then using SuspendWorkflow() when it was doing step b.
in my opinion, .SuspendWorkflow() can suspend the running workflow.so the step c & d will be executed after ResumeWorkflow() down.

...
var res = _workflowHost.SuspendWorkflow(_workflowId).Result;
// res value is false
...

but the real operation is the workflow running all steps and SuspendWorkflow() return false always.
what`s wrong?

**Comment by hblwslyan-commits** [comment]
![Uploading Screenshot_٢٠٢٥-١١-٢٥-١٠-٠٥-٣٨-٥٨٢_com.instagram.android-edit.jpg…]()
